Character.cpp

```diff
@@ -98,20 +99,48 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        //a)
+        if( hitPoints < *initialHitPoints )
+        {
+            hitPoints = *initialHitPoints;
+        }
+        if( armor < *initialArmorLevel )
+        {
+            armor = *initialArmorLevel;
+        }
+        if ( attackDamage < *initialAttackDamage )
+        {
+            attackDamage = *initialAttackDamage;
+        }
+        //b)
+        hitPoints *= 1.1;
+        armor *= 1.1;
+        attackDamage *= 1.1;
+        //c)
+        initialHitPoints.reset( new int(hitPoints) );
+        initialArmorLevel.reset( new int( armor) );
+        initialAttackDamage.reset( new int( attackDamage) );
```

Comment:
You're doing the same thing 3 times over.
D.R.Y.
Don't Repeat Yourself
refactor this into a function that takes two references to `int`'s. 
Don't use `new` to update the `initial` objects.  Just dereference them
- [ ] revise